lib/hexpm/billing/billing.ex

```diff
@@ -22,4 +24,16 @@ defmodule Hexpm.Billing do
   def invoice(id), do: impl().invoice(id)
   def pay_invoice(id), do: impl().pay_invoice(id)
   def report(), do: impl().report()
+
+  def create(params, audit: audit_data) do
+    Multi.new()
+    |> Multi.run(:create, fn _, _ -> impl().create(params) end)
+    |> audit(audit_data, "billing.create", params)
+    |> Repo.transaction()
```

Comment:
I don't think we should use `Multi` here since we are only doing a single action against the database.

Reply:
Totally agreed!

I used `Multi` here because `audit/4` required a `Multi`.

If we are not going to use `Multi`, maybe I should add an `AuditLog.audit` function that doesn’t require `Multi`?

Reply:
Oh didn't think of that. Adding a function that doesn't use `Multi` sounds great!

Reply:
Okay, I’ll open a PR for that.

Reply:
Please do it in this PR if possible so we can track the changes together.

Reply:
`AuditLog.audit/3` is added in this PR (https://github.com/hexpm/hexpm/pull/837/commits/b540635f399325035d24169f3afc52be4583037c)